src/python/pants/notes/master.rst

```diff
@@ -4,6 +4,140 @@ Master Pre-Releases
 This document describes development releases which occur weekly from master, and which have
 not undergone the vetting associated with ``stable`` releases.
 
+1.25.0.dev3 (01/29/2020)
+------------------------
+
+API Changes
+~~~~~~~~~~~
+
+* Remove the generate-pants-ini goal. (#8993)
+  `PR #8993 <https://github.com/pantsbuild/pants/pull/8993>`_
+
+* Deprecate `--fast` option (#8970)
+  `PR #8970 <https://github.com/pantsbuild/pants/pull/8970>`_
+
+* Switch to dot idea project type (#8745)
+  `PR #8745 <https://github.com/pantsbuild/pants/pull/8745>`_
+
+* Deprecate defaulting to `--transitive` (#8955)
+  `PR #8955 <https://github.com/pantsbuild/pants/pull/8955>`_
+
+New Features
+~~~~~~~~~~~~
+
+* Allow excludes in BUILD files via new `!` syntax (#9025)
+  `PR #9025 <https://github.com/pantsbuild/pants/pull/9025>`_
+
+* Refactor `PathGlobs` to support `!` ignore globs in its constructor (#9032)
+  `PR #9032 <https://github.com/pantsbuild/pants/pull/9032>`_
+
+* Modify options parser to allow `!` in file args (#9031)
+  `PR #9031 <https://github.com/pantsbuild/pants/pull/9031>`_
```

Comment:
Also a refactoring. This was necessary to land #9025 but not a new user-facing feature on its own.